packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -22,10 +22,12 @@ exports.sourceNodes = async (
     },
   })
 
-  // Touch existing Contentful nodes so Gatsby doesn't garbage collect them.
+  // Delete existing Contentful nodes since we are doing a bulk re-add
+  // Once we are sync'ing Contentful nodes to get the diff, this would
+  // need to be changed.
   _.values(store.getState().nodes)
-    .filter(n => n.internal.type.slice(0, 8) === typePrefix)
-    .forEach(n => touchNode(n.id))
+    .filter(n => n.internal.type.indexOf(typePrefix) === 0)
```

Comment:
oh sorry this isn't needed actually either. Gatsby will "garbage collect" any node on startup that isn't either created or touched. Since you're creating (again) every node then we're fine. If something *has* been deleted on the Contentful side, than it just won't be created again which means Gatsby will kill it. So if we're not syncing the logic is pretty simple. Just create everything. When syncing then you have to worry about telling Gatsby what is and isn't still around.